### PR TITLE
polish(incremental): rename terms to match spec

### PR DESCRIPTION
**Pull request by yaacovCR**
DeferredGroupedFieldSet => ExecutionGroup
SubsequentResultRecord => DeliveryGroup
ExecuteExecutionGroups => CollectExecutionGroups

### CODE REVIEW

**Review by JoviDeCroock — APPROVED** [review]

**Comment by yaacovCR** [comment]
I should make clear that this PR is to match the spec draft WIP at https://github.com/graphql/graphql-spec/pull/1077 and not the actual working draft of the spec, which does not yet have incremental delivery at all.

File: src/execution/IncrementalGraph.ts [hunk 1]
```diff
@@ -49,23 +46,25 @@ export class IncrementalGraph {
     return this._promoteNonEmptyToRoot(initialResultChildren);
   }
 
-  addCompletedReconcilableDeferredGroupedFieldSet(
-    reconcilableResult: ReconcilableDeferredGroupedFieldSetResult,
+  addCompletedReconcilableExecutionGroup(
```
Comment (robrichard) [inline src/execution/IncrementalGraph.ts:49]: should this function be renamed to `addCompletedSuccessfulExecutionGroup`?

File: src/execution/execute.ts [hunk 2]
```diff
@@ -296,7 +296,7 @@ function executeOperation(
         undefined,
       );
     } else {
-      const fieldPLan = buildFieldPlan(groupedFieldSet);
+      const fieldPLan = buildExecutionPlan(groupedFieldSet);
```
Comment (robrichard) [inline src/execution/execute.ts:299]: rename `fieldPLan` to `executionPlan`?

File: src/execution/IncrementalGraph.ts (hunk 1 above)
Comment (yaacovCR) [reply src/execution/IncrementalGraph.ts]: Yes!

db4482a6

File: src/execution/execute.ts (hunk 2 above)
Comment (yaacovCR) [reply src/execution/execute.ts:299]: done 1f44a7ca

**Review by robrichard — APPROVED** [review]

**Pull request merged**